Providers/Modules/Plugins/ChangeTracking/plugin/changetracking_lib.rb

```diff
@@ -281,4 +281,60 @@ def self.setHash(prev_hash, last_upload_time, file_path)
                         File.write(file_path, "#{PREV_HASH}=#{prev_hash}\n#{LAST_UPLOAD_TIME}=#{last_upload_time}")
                 end
     end
+
+
+    def self.transform_and_wrap(inventoryFile, inventoryHashFile)
+		if File.exist?(inventoryFile)                       
+			@@log.debug ("Found the change tracking inventory file.")
+			# Get the parameters ready.
+			time = Time.now
+			force_send_run_interval_hours = 24
+			force_send_run_interval = force_send_run_interval_hours.to_i * 3600
+			@hostname = OMS::Common.get_hostname or "Unknown host"
+
+			# Read the inventory XML.
+			file = File.open(inventoryFile, "rb")
+			xml_string = file.read; nil # To top the output to show up on STDOUT.
+
+			previousSnapshot = ChangeTracking.getHash(inventoryHashFile)
+                        previous_inventory_checksum = {}
+                        begin
+                          if !previousSnapshot.nil?
+                            previous_inventory_checksum = JSON.parse(previousSnapshot[PREV_HASH])
+                          end
+                        rescue 
+			@@log.warn ("Error parsing previous hash file")
+                             previousSnapshot = nil
+                        end
+			#Transform the XML to HashMap
+			transformed_hash_map = ChangeTracking.transform(xml_string, @@log)
+                        current_inventory_checksum = ChangeTracking.computechecksum(transformed_hash_map)
+                        changed_checksum = ChangeTracking.comparechecksum(previous_inventory_checksum, current_inventory_checksum)
+                        transformed_hash_map_with_changes_marked = ChangeTracking.markchangedinventory(changed_checksum, transformed_hash_map)
+
+			output = ChangeTracking.wrap(transformed_hash_map_with_changes_marked, @hostname, time)
+			hash = current_inventory_checksum.to_json
+
+			# If there is a previous hash
+			if !previousSnapshot.nil?
+				# If you need to force send
+                                previousSnapshotTime = DateTime.parse(previousSnapshot[LAST_UPLOAD_TIME]).to_time
+				if force_send_run_interval > 0 and 
+					Time.now.to_i - previousSnapshotTime.to_i > force_send_run_interval
+					ChangeTracking.setHash(hash, Time.now,inventoryHashFile)
+				elsif !changed_checksum.nil? and !changed_checksum.empty?
+					ChangeTracking.setHash(hash, Time.now, inventoryHashFile)
+				else
+					return {}
+				end
+			else # Previous Hash did not exist. Write it
+				# and the return the output.
+				ChangeTracking.setHash(hash, Time.now, inventoryHashFile)
+			end
+			return output
+		else
+			@@log.warn ("The ChangeTracking File does not exists. Make sure it is present at the correct")
```

Comment:
This sentence has been incomplete for a long time. I'm assuming you meant to say "The ChangeTracking File does not exists. Make sure it is present at the correct location." ?